### DISCUSSION TITLE
I can't properly aggregate data into array

### DISCUSSION THREAD

**Discussion by Skipperlla** [Shopify/flash-list#597] [Q&A]
In my current error, I keep the length of the button in an array and show how many cards you have selected in parenthesis in total, but as you can see in the video, when I want to select one of the cards and remove it, I cannot do this.

https://user-images.githubusercontent.com/68515357/189965711-27029045-319e-4a94-b6d0-8e5946a2c2a6.mp4

**Comment by fortmarek** [comment]
Hey!

Please post a reproducible sample or at least a code snippet, it's hard to help without that 😞

**Reply by Skipperlla** [reply]
This code my flashlist
```js   
<FlashList
            data={Media?.results}
            renderItem={RenderItem}
            numColumns={2}
            getItemType={(item) => {
              return item.id;
            }}
            keyExtractor={keyExtractor}
            onEndReachedThreshold={0.5}
            estimatedItemSize={200}
            onEndReached={onEndReached}
            ListFooterComponent={ListFooterComponent}
          />
```
Here codes my renderItem
```js   
 const RenderItem = useCallback(
    ({item, index}: {item: IMedia; index: number}) => (
      <MediaCard
        item={item}
        photo={item?.posterPhoto || item?.profilePhoto || item?.photo}
        originalTitle={item?.originalTitle || item?.originalName}
        title={item?.name || item?.title}
        id={item.id}
        index={index}
        key={index}
        score={item.score}
        genres={item?.genres}
        selectedMedia={selectedMedia}
        mediaType={item.mediaType}
        queryMediaType={query.mediaType}
        setSelectedMedia={setSelectedMedia}
        selectedCard={
          selectedMedia?.filter((result) => {
            return item.id === result.id;
          })?.length
        }
      />
    ),
    [selectedMedia],
  );
```
what I'm doing is const [selected Media, setSelected Media] = useState<Media[]>([]); I have a state called and I save every card I print in it, but as you can see in the video, it doesn't do that.

**Comment by Skipperlla** [comment]
when i try to do the same with flatlist it works even this video

https://user-images.githubusercontent.com/68515357/190156750-1e9eaa5a-0e11-4329-958f-c7c229fc8b01.mp4

If you notice in the video, the place with the parenthesis below beats every time I press it, so I can save it to the state.
